jobs/delegatesActions.js

```diff
@@ -0,0 +1,50 @@
+const Constants = require("../constants/Constants");
+const IssuerService = require("../services/IssuerService");
+const { ERROR, DONE, ERROR_RENEW } = Constants.STATUS;
+const { CREATE, REFRESH, REVOKE } = Constants.DELEGATE_ACTIONS;
+
+const exCallback = async ({ callbackUrl, did, token, status = ERROR, expireOn, blockHash, messageError }) => {
+	await IssuerService.callback(callbackUrl, did, token, { status, expireOn, blockHash, messageError });
+};
+
+const catchError = async (next, dataError) => {
+	try {
+		await next();
+	} catch (err) {
+		console.log(err);
+		const messageError = err.message ? err.message : err;
+		exCallback({ ...dataError, messageError });
+	}
+};
+
+const funcToDone = (next, data, statusError) => {
+	catchError(
+		async () => {
+			const { blockHash, expireOn } = await next(data);
+			exCallback({ ...data, status: DONE, expireOn, blockHash });
+		},
+		{ ...data, status: statusError }
+	);
```

Comment:
no me cierra mucho el catchError, si te fijas, este mismo fragmento podria quedar sin el catchError asi:
```suggestion
       try {
            const { blockHash, expireOn } = await next(data);
            exCallback({ ...data, status: DONE, expireOn, blockHash });
       } catch(err) {
            handleError(err)  // en esta funcion podes poner la forma en que manejas los errores
       }
```
Y la funcion se explica mucho mejor por si sola